### ISSUE TITLE
makeidx improperly referencing page numbers. 

### ISSUE THREAD

**Issue by jkylearmstrong** [quarto-dev/quarto-cli#4585]
### Bug description

```{r}
RStudio 2023.03.0-daily+70 "Cherry Blossom" Daily (236fd07b9b86d184ae614fffe20e44f8f2b1937b, 2023-01-09) for Windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.03.0-daily+70 Chrome/108.0.5359.62 Electron/22.0.0 Safari/537.36
```

For some reason my PDF is improperly referencing indexes made with the index at the end.  The index will have a link to page 55 but the text is at a different page?

```{r}
  pdf:
    documentclass: krantz
    include-in-header: 
      text: |
        \usepackage{makeidx}
        \makeindex
        \usepackage{fvextra}
        \DefineVerbatimEnvironment{Highlighting}{Verbatim}{breaklines,commandchars=\\\{\}}
        \DefineVerbatimEnvironment{OutputCode}{Verbatim}{breaklines,commandchars=\\\{\}}
    include-after-body: 
      text: |
        \printindex
    lof: true
    lot: true
    prefer-html: true
    toc: true
    number-sections: true
    colorlinks: true
    toc-depth: 5
    keep-md: false
    keep-tex: true
    latex-min-runs: 4
    latex-max-runs: 7

crossref:
  chapters: true
  lof-title: "List of Figures"
  lot-title: "List of Tables"
```

### Checklist

- [ ] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
Thanks for the report. Unfortunately, we need a full example to be able to diagnose the problem (and, ideally, a small one, although the fact that you said "page 55" makes me think it won't be small...)

**Comment by jkylearmstrong** [comment]
I am going to see if I can create an example with adding some text and other items to the standard quarto book outline to see if I can get the same error, however, I am not able to share my repo for the book I am working on.

**Comment by cscheid** [comment]
If it's something you're comfortable sharing in private (and we'd make sure not to add any content of it in a test), you can reach me over email at carlos.scheidegger@posit.co

**Comment by jkylearmstrong** [comment]
I found a similar problem and potential solution on: [https://github.com/rstudio/bookdown/issues/1274#top](https://github.com/rstudio/bookdown/issues/1274#top) this appears to have addressed with bookdown however, is there anyway to get similar results with Quarto? I have also posted to the bookdown git to see if anyone over there has suggestions : [https://github.com/rstudio/bookdown/issues/1411](https://github.com/rstudio/bookdown/issues/1411)

**Comment by cscheid** [comment]
@jkylearmstrong I'll only be able to fully diagnose this when I can repro. Did you see my previous message? Thank you!

**Comment by cderv** [comment]
@jkylearmstrong thanks for finding the **bookdown** issue. We indeed needed a fix in **tinytex** R package related to how `makeindex` works. Fix is here for reference: https://github.com/rstudio/tinytex/commit/5d211d43944d322fca49e5f0d97f34b9c46ff9ab and was identified in [bookdown](https://github.com/rstudio/bookdown/issues/1274#issuecomment-1020475236)

Quarto is not using the **tinytex** R package but logic is the same. @dragonstyle we should check that we are running LaTeX engine the same number of time.

**Comment by jkylearmstrong** [comment]
@dragonstyle and @cderv - is there anyway for me to test?

**Comment by cderv** [comment]
Yes - the fix should be included in [v1.3.247](https://github.com/quarto-dev/quarto-cli/releases/tag/v1.3.247)

you can get it from https://quarto.org/docs/download/prerelease.html

**Comment by jkylearmstrong** [comment]
Thank you all! This seems to have solved the issue!
